src/main/java/br/ufpe/cin/app/JFSTMerge.java

```diff
@@ -60,7 +60,15 @@
 	
 	@Parameter(names = "-l", description = "Parameter to disable logging of merged files (true or false).",arity = 1)
 	public static boolean logFiles = true;
-
+	
+	@Parameter(names = "-ignore-space-change", description = "Treats lines with the indicated type of whitespace change as unchanged for "
+			+ "the sake of a three-way merge. Whitespace changes mixed with other changes to a line are not ignored.", arity = 1)
+	public static boolean isWhitespaceIgnored = true;
+	
+	@Parameter(names = "-encoding-inference", description = "Tries to infer file encodings to properly merge them."
+			+ " If not enabled, the tool assumes files are encoded in UTF-8.", arity = 1)
+	public static boolean isEncodingInferenceEnabled = true;
```

Comment:
retira o "arity = 1" como em "isGit"
a aridade obriga a colocar true|false junto ao parametro, exemplo "-encoding-inference true"
sem a aridade, ele pega o valor default

Reply:
É, mas de acordo com a documentação, se o valor default dele for true, ele tem que ter aridade 1, porque senão não dá pra falsear (e o programa gera uma exceção em tempo de execução caso tente). Se for pra ativar ele, não é necessário coloca-lo e, se colocar, não precisa explicitar o true. Mas se é pra desativar, tem que explicitar o false.

Reply:
Tu conseguiu confirmar isso dessa exceção? acho que isso só acontece com parâmetros não booleanos. Digo isso porque da experiência com os outros parâmetros, sem aridade, quando não colocamos o parâmetro, ele pega o valor default. Caso queira inverter o valor default, ai sim colocamos o "-isEncodingInferenceEnabled". 

Resumindo, com aridade, caso desejemos colocar como false, tem que fazer:
java -jar s3m.jar -isEncodingInferenceEnabled false

Sem aridade basta:
java -jar s3m.jar -isEncodingInferenceEnabled

Reply:
É porque todos os parâmetros com aridade 0 são false por default, por isso funciona.
Quando é true por default, tanto faz colocar o parâmetro ou não (colocar o parâmetro deixa ele true, não troca o valor default). Isso é o que diz a documentação (e eu testei agora pra confirmar):
_"If you want to define a boolean parameter that’s true by default, you can declare it as having an arity of 1. Users will then have to specify the value they want explicitly"_ (http://jcommander.org/)

Nota que os parâmetros _isCryptographed_ e _logFiles_ seguem o mesmo padrão, e eles já existiam antes. Claro, eu posso deixar o parâmetro false por default, mas creio que perderia o sentido.

Reply:
Este é o problema: "Users will then have to specify the value they want explicitly". Mais uma fonte de erro pro usuário.

Na minha experiência, sem aridade, eu consigo manipular os valores do parametro apenas colocando ou removendo ele, sem precisar especificar true|false. Dito isso, eu acho que tem como tu deixar ```isEncodingInferenceEnabled``` com default ```true```, sem aridade. Faz o teste se funciona.

Reply:
@guilhermejccavalcanti pelo que entendi de "Quando é true por default, tanto faz colocar o parâmetro ou não", já está assim. não @jvcoutinho?

Reply:
@pauloborba sim, mas o que @guilhermejccavalcanti diz é que o argumento do parâmetro (true ou false) é dispensável se ele for true por default. Porém, como visto neste teste e como previsto na documentação:
![2018-09-14](https://user-images.githubusercontent.com/27645414/45550622-c1297d80-b801-11e8-99d2-cb2295c82d35.png)
Não é possível falsear o parâmetro se ele for true por default com aridade 0. Aridade 0 só funciona se ele for false por default.

Reply:
Tem alguma coisa diferente. Olha o exemplo do isGit:

``` @Parameter(names = "-g", description = "Parameter to identify that the tool is being used as a git merge driver.")
public static boolean isGit = false; ```

E olha como eu chamo ele:

![image](https://user-images.githubusercontent.com/3333981/45554002-b96ed680-b80b-11e8-837d-b9c8759d8fc7.png)

Não tem aridade, o default é false, e eu consigo inverter o valor apenas adicionando o parametro.

Reply:
Tem alguma coisa diferente. Olha o exemplo do isGit:

@Parameter(names = "-g", description = "Parameter to identify that the tool is being used as a git merge driver.")
public static boolean isGit = false; 

E olha como eu chamo ele:

![image](https://user-images.githubusercontent.com/3333981/45554002-b96ed680-b80b-11e8-837d-b9c8759d8fc7.png)

Não tem aridade, o default é false, e eu consigo inverter o valor apenas adicionando o parametro.

Reply:
Exatamente, o default do isGit é false. Por isso funciona.
Adicionar o parâmetro não inverte o valor default, e sim torna o parâmetro true.

Reply:
Entendi, que bad. Tenho usado outro "serviço" de parametros numa outra ferramenta e é bem mais simples. Mas enfim, prioridade minima isso.